### ISSUE TITLE
build-for-testing and CI/CD

### ISSUE THREAD

**Issue by im-jersh** [SlatherOrg/slather#531]
i have a CI/CD pipeline where one job builds my project for testing and then a subsequent job runs the tests with the generated `.xctestrun` from the build job. this pipeline optimization has worked great for me. however, i run into issues when trying to use Slather.

due to jobs being distributed across a pool of runners, if the build and test jobs run on different machines, Slather does not generate any reports because it can't find the files used to build. it looks like Slather is using the absolute file paths encoded in the test results output which references paths that are only valid on the runner that executed the build job, which of course don't exist on the runner that executed the test and report job.

how does one work around this? is there a way to provide the source directory rather than rely on the absolute file paths encoded in the test results? or am i to resort to just executing my build and test commands in the same job?

**Comment by bratxdy** [comment]
Did you end up figuring this out? I ran into the same issues on Bitrise.

**Comment by im-jersh** [comment]
nope, but i haven't really spent much time trying to figure this out since opening this issue.

i imagine this likely has more to do with the internal structuring of the test result artifacts but i'm still hopeful there's a way Slather could work around this if that holds true.
